### PR TITLE
Improve Games filtering and sorting, improve dev and travis docker configurations

### LINKED ISSUE #132: Add "Has a demo" to search filter
Add the option to filter in "has a demo" for search results.

### LINKED ISSUE #148: Games list sorting
Hello,
a function that sorts the list by features like popularity, date added, wine quality, or similar would be great.

### LINKED ISSUE #164: The Platforms and Genres Filters are cluttered, small, and hard to read
It uses small white on grey text, the items are in rows of seemingly random length, the boxes appear cluttered, and it is hard to visually scan for the one you want.  

I recommend making the buttons at the very least a darker background color.

EDIT: Sugestion: use a tree based system which only shows the largest catagories.  
IE: Action 
       shooter 
         third person shooter

### LINKED ISSUE #186: Search in game aliases
Currently, if you search for "GTA V" or even just plain GTA, it will return 0 results, even though there are plenty of Grand Theft Auto installers on Lutris. 
This may confuse the user that Lutris doesn't have installers for these or other games with similar naming quirks (like Civilization - Civ).
I suggest to either make search show up games that match the search value not only with just the title, but with words from description too, or add search keywords for game pages so contributors could add them manually.

### LINKED ISSUE #198: Allow genre and platform conditions to be applied together in search
Hi,

The concept of Lutris is great, but there is something I'd like to request. Could you's please allow us to apply platform and genre conditions together? Like presently I'd like to search for Linux-compatible RTS games only, but I can only search for [RTS games](https://lutris.net/games/genre/rts/) (in which RTS games compatible with any platform will appear) OR [Linux-compatible games](https://lutris.net/games/platform/linux/) (in which Linux-compatible games of all genres will appear). 

Thanks for your time,
Brenton

### LINKED ISSUE #200: Search function not showing proper results for individual words
Ended up submitting a new game by accident due to the search function not finding it even though it exists. Typing "tiny big" showed nothing even though there's a game called "Tiny and Big: Grandpa's Leftovers" already, which resulted in me making a new entry and not realizing it existed until I tried forking a GOG installer and saw both the entry I created and the entry that already existed (I used "&" instead of "and" for the title to be more accurate, which is why no alert showed up about the game already existing).

Not the only game it happened to, either: As an example, typing "night woods" shows nothing, while typing "night in the woods" shows the actual game.

### LINKED ISSUE #253: Can't filter search results by platform
How to reproduce:
1. Open Games section: https://lutris.net/games/
2. Select a platform (for example, Windows): https://lutris.net/games/platform/windows
3. Type in the name of game into `Search` box
4. Get 404 error: https://lutris.net/games/platform/?q=The+Sims

### LINKED ISSUE #380: Display hint that results contain unpublished entries when searching for games
Hello,
it's not the first time I had this issue on Lutris website: I searched for a game on Lutris site that gave me no result. So I decided to add this new game to the database.
I filled in the different fields of the form and click on "submit" and...I obtained this message: the game is already in our database...I loose my time several times on this issue. Example: Alone in the Dark 2, Age of Empire (first title),  etc...

### LINKED ISSUE #395: Search function should ignore ":"
`Red Faction: Guerrilla` isn't found when searching `Red Faction Guerrilla`

https://lutris.net/games/?q=Red+Faction+Guerrilla

https://lutris.net/games/add-game/games/red-faction-guerrilla/

### PR DESCRIPTION
**Pull request by AlexanderRavenheart**
- Resolves #132
- Resolves #380 (added count of unpublished games next to total number of entries)
- Resolves #198
- Resolves #186
- Resolves #395
- Resolves #200
- Resolves #253
- Resolves #148
- Resolves #164

### CODE REVIEW

**Comment by AlexanderRavenheart** [comment]
Can this be merged or are there any issues I missed?

**Comment by AlexanderRavenheart** [comment]
The native build fails because distinct('field') (DISTINCT ON) is only supported on PostgreSQL. Native build uses SQLite.

**Comment by AlexanderRavenheart** [comment]
Build issues solved. PR ready for review.

File: Makefile [hunk 1]
```diff
@@ -74,3 +60,10 @@ sync:
 	gunzip lutris.tar.gz
 	pg_restore -h localhost -U lutris --clean --dbname=lutris lutris.tar
 	rm lutris.tar
+
+dev_docker:
+	mkdir -p ./src
+	git clone https://github.com/strycore/django-croppie.git ./src/django-croppie
```
Comment (strycore) [inline Makefile:66]: What is happening here? I've never had to do this… why isn't Pip able to install those normally?

File: config/requirements/base.pip [hunk 2]
```diff
@@ -17,13 +17,10 @@ django-rest-swagger==2.2.0
 django-reversion==3.0.5
 django-bitfield==2.0.1
 django-webpack-loader==0.7.0
-# The production release does resizing on the backend and doesn't support
-# updates
-# django-croppie==1.3
--e git+https://github.com/strycore/django-croppie.git#egg=django-croppie
```
Comment (strycore) [inline config/requirements/base.pip:23]: Why did you remove the git ones?

File: config/requirements/devel.pip [hunk 3]
```diff
@@ -1,9 +1,15 @@
 -r base.pip
 # django-jenkins is unmaintained, the lutris fork contains updates
 # to support current versions of Coverage and Pylint
--e git+https://github.com/lutris/django-jenkins.git#egg=django-jenkins
+-e ../../src/django-jenkins
+# The production release does resizing on the backend and doesn't support
+# updates
+# django-croppie==1.3
+-e ../../src/django-croppie
```
Comment (strycore) [inline config/requirements/devel.pip:8]: What?

File: config/requirements/devel.pip [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
+-e ../../src/django-openid-auth
```
Comment (strycore) [inline config/requirements/devel.pip:9]: Again, WTF?

File: config/requirements/production.pip [hunk 5]
```diff
@@ -1,4 +1,9 @@
 -r base.pip
+# The production release does resizing on the backend and doesn't support
+# updates
+# django-croppie==1.3
+git+https://github.com/strycore/django-croppie.git#egg=django-croppie
```
Comment (strycore) [inline config/requirements/production.pip:5]: Oh now they're back!

File: dev.Dockerfile [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+ENV LC_ALL=C.UTF-8
+ENV SECRET_KEY="somethissecret"
+ENV DB_HOST="lutrisdb"
+ENV DJANGO_SETTINGS_MODULE="lutrisweb.settings.local"
+ENV REDIS_HOST='lutriscache'
+
+RUN apt-get update && apt-get install -y sudo build-essential git curl python3 \
+    python3-pip python3-dev imagemagick libxml2-dev libxslt1-dev libssl-dev libffi-dev \
+    libpq-dev libxml2-dev libjpeg-dev
+RUN curl -sL https://deb.nodesource.com/setup_13.x | bash -
```
Comment (strycore) [inline dev.Dockerfile:15]: Remove, the node version shipped with 20.04 is good enough

File: dev.Dockerfile [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+RUN apt-get install -y nodejs
```
Comment (strycore) [inline dev.Dockerfile:16]: Remove too, just add npm to the list of system dependencies

File: dev.vue.Dockerfile [hunk 8]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu:20.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+
+ENV LC_ALL=C.UTF-8
+
+RUN apt-get update && apt-get install -y --no-install-recommends sudo build-essential git curl ca-certificates
+RUN curl -sL https://deb.nodesource.com/setup_13.x | bash -
```
Comment (strycore) [inline dev.vue.Dockerfile:8]: Just use a node Docker image here

Comment (strycore) [reply dev.vue.Dockerfile:8]: If possible, it seems it's just Building the vue project

File: lutrisweb/settings/local.py [hunk 9]
```diff
@@ -15,8 +15,8 @@
             'NAME': os.environ.get("POSTGRES_DB", "lutris"),
             'USER': os.environ.get("POSTGRES_USER", "lutris"),
             'PASSWORD': os.environ.get('POSTGRES_PASSWORD', "admin"),
-            'HOST': os.environ.get('POSTGRES_HOST', 'localhost'),
-            'PORT': os.environ.get('POSTGRES_PORT', '5432'),
+            'HOST': os.environ.get('POSTGRES_HOST', 'lutrisdb'),
+            'PORT': os.environ.get('POSTGRES_PORT', '5430'),
```
Comment (strycore) [inline lutrisweb/settings/local.py:19]: This should stay the default port. Change it in your Dockerfile.

File: lutrisweb/settings/local.py [hunk 10]
```diff
@@ -15,8 +15,8 @@
             'NAME': os.environ.get("POSTGRES_DB", "lutris"),
             'USER': os.environ.get("POSTGRES_USER", "lutris"),
             'PASSWORD': os.environ.get('POSTGRES_PASSWORD', "admin"),
-            'HOST': os.environ.get('POSTGRES_HOST', 'localhost'),
```
Comment (strycore) [inline lutrisweb/settings/local.py:18]: This should stay localhost

File: lutrisweb/settings/test.py [hunk 11]
```diff
@@ -2,11 +2,32 @@
 
 SEND_EMAILS = False
 DEBUG = False
-DATABASES = {
-    'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': ':memory:',
+if os.getenv('USE_SQLITE', '0') == '1':
```
Comment (strycore) [inline lutrisweb/settings/test.py:5]: If there's a way to completely remove support for SQLite, then I'd be in favor of getting rid of it

File: Makefile (hunk 1 above)
Comment (AlexanderRavenheart) [reply Makefile:66]: I noticed it fails to install them in the python site-packages even though it says they are installed. Probably a linking error.
Anyway, this way makes much more sense in dev environment to be able to see live changes if one wants to update the source of one of those projects.

File: config/requirements/base.pip (hunk 2 above)
Comment (AlexanderRavenheart) [reply config/requirements/base.pip:23]: The git ones are to be used in prod only. In dev you may want to make changes to those projects and see the changes reflected locally thus having the source makes sense.

File: lutrisweb/settings/local.py (hunk 9 above)
Comment (AlexanderRavenheart) [reply lutrisweb/settings/local.py:19]: You requested this change to not conflict with the local install of Postgres.

Comment (strycore) [reply lutrisweb/settings/local.py:19]: I meant in the Dockerfile. The Django project should still be able to find a standard postgres server if ran natively.

File: config/requirements/devel.pip (hunk 3 above)
Comment (AlexanderRavenheart) [reply config/requirements/devel.pip:8]: Explained in Discord.

File: config/requirements/devel.pip (hunk 4 above)
Comment (AlexanderRavenheart) [reply config/requirements/devel.pip:9]: Same as above.

File: dev.Dockerfile (hunk 6 above)
Comment (AlexanderRavenheart) [reply dev.Dockerfile:15]: Okay, will do.

File: dev.Dockerfile (hunk 7 above)
Comment (AlexanderRavenheart) [reply dev.Dockerfile:16]: Okay.

File: dev.vue.Dockerfile (hunk 8 above)
Comment (AlexanderRavenheart) [reply dev.vue.Dockerfile:8]: Yep, gonna try that next.

File: lutrisweb/settings/local.py (hunk 10 above)
Comment (AlexanderRavenheart) [reply lutrisweb/settings/local.py:18]: No it shouldn't because the service is run in a docker and to avoid a security issue the db port is only exposed to other docker services.

File: lutrisweb/settings/test.py (hunk 11 above)
Comment (AlexanderRavenheart) [reply lutrisweb/settings/test.py:5]: I will see what can be done to completely remove SQLite support.

File: lutrisweb/settings/local.py (hunk 9 above)
Comment (AlexanderRavenheart) [reply lutrisweb/settings/local.py:19]: Okay, then I will add the ENV variables to the docker container to use lutrisdb and 5430 and change this to use localhost and 5432 by default.

File: lutrisweb/settings/local.py (hunk 10 above)
Comment (AlexanderRavenheart) [reply lutrisweb/settings/local.py:18]: Will make the changes mentioned above with the ENV variables.
